### DISCUSSION TITLE
Handshake modification

### DISCUSSION THREAD

**Discussion by okbaswc** [microsoft/msquic#2184] [Q&A] [answered]
I'm having trouble finding the flow in and out of UDP messages during the initial handshake. I'd like to find where UDP messages are coming in and are decrypted so I can print out the frames and payload, and also where responses are sent so I can print those out as well, to understand more fully what's going on and to modify some of the responses.

Where is this happening in the code base?

**Comment by nibanks** [comment, marked as answer]
We have pretty good logging for this already. You can follow the instructions [here](https://github.com/microsoft/msquic/blob/main/docs/Diagnostics.md#trace-collection) to get a trace. Then take a look at `[TX]` and `[RX]` traces. In the code, you can look for instances of [QuicPacketLogHeader](https://github.com/microsoft/msquic/blob/main/src/core/packet.c#L532) and [QuicFrameLogAll](https://github.com/microsoft/msquic/blob/main/src/core/frame.c#L1875).
